middlewares/ip_whitelister_test.go

```diff
@@ -0,0 +1,302 @@
+package middlewares
+
+import (
+	"errors"
+	"fmt"
+	"github.com/codegangsta/negroni"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+func TestNewIPWhitelister(t *testing.T) {
+	cases := map[string]struct {
```

Comment:
In my mind the tests are grouped by name, just as you would do it when writing them down on paper. You would not have a Headline "1" with the title as Content under it. But I'm ok following your style guide and will change this :)